### PR TITLE
Feature box plot

### PR DESCRIPTION
**Pull request by caruta**
## Description
A new 2D histogram containing the values of noise for each VFATs has been added. It is drawn as a box plot in order to facilitate the visualization of the main characteristics of the noise distributions.

### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Motivation and Context
It helps the visualization of the main characteristics of the noise distribution for each VFAT.

## How Has This Been Tested?
It has been tested comparing the results of two scurves taken in different conditions: HV ON & HV OFF.

### Screenshots (if appropriate):
canvscurvesigma_boxplot

## Checklist:
- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: anaUltraScurve.py [hunk 1]
```diff
@@ -795,6 +795,17 @@ def plotAllSCurvesOnCanvas(vfatHistos, vfatHistosPanPin2=None, obsName="scurves"
         saveSummary(effPedSummaryPlots, None, '%s/ScurveEffPedSummary.png'%filename, None, drawOpt="E1")
         saveSummary(encSummaryPlots, None, '%s/ScurveSigmaSummary.png'%filename, None, drawOpt="AP")
 
+        #BoxPlot
+        canvasBoxPlot = r.TCanvas("h2ENC","h2ENC",0,0,1200,1000)
+        h2DetENC_All.SetStats(0)
+        h2DetENC_All.GetXaxis().SetTitle("VFAT position")
+        h2DetENC_All.GetYaxis().SetTitle("Noise (fC)")
```
Comment (bdorney) [inline anaUltraScurve.py]: if you do `Noise #left(fC#right)` you'll have parenthesis properly sized for whatever is present, see for example:

https://root.cern.ch/doc/master/classTLatex.html#L6

File: macros/plotSCurveFitResults.py [hunk 2]
```diff
@@ -383,9 +394,19 @@
 
         canvScurveEffPed_DetSum.cd()
         if idx == 0:
-            dict_ScurveEffPed[chamberAndScanDatePair][-1].Draw("E1")
+	    dict_ScurveEffPed[chamberAndScanDatePair][-1].Draw("E1")
```
Comment (bdorney) [inline macros/plotSCurveFitResults.py]: looks like your editor has slipped in a `\t` character still.

File: anaUltraScurve.py (hunk 1 above)
Comment (caruta) [reply anaUltraScurve.py]: Ok, I will modify it

File: macros/plotSCurveFitResults.py [hunk 3]
```diff
@@ -383,9 +394,19 @@
...
         else:
             dict_ScurveEffPed[chamberAndScanDatePair][-1].Draw("sameE1")
+	
+	canvScurveSigma_boxPlot.cd()
+	dict_ScurveSigma_boxPlot[chamberAndScanDatePair].SetFillColorAlpha(getCyclicColor(idx), 0.3)
+	dict_ScurveSigma_boxPlot[chamberAndScanDatePair].SetLineColor(getCyclicColor(idx))
+	if idx == 0:
+	    dict_ScurveSigma_boxPlot[chamberAndScanDatePair].GetXaxis().SetTitle("VFAT position")
+            dict_ScurveSigma_boxPlot[chamberAndScanDatePair].GetYaxis().SetTitle("Noise (fC)")
+	    dict_ScurveSigma_boxPlot[chamberAndScanDatePair].Draw("candle1")
```
Comment (bdorney) [inline macros/plotSCurveFitResults.py]: Again here, your editor probably is using whitespace that does not match.  If you where to use `vim` the relevant line in your `$HOME/.vimrc` to convert tab's to spaces is:

```
filetype plugin indent on
" show existing tab with 4 spaces width
set tabstop=4                                                                                                                                                                                                                                
" when indenting with '>', use 4 spaces width
set shiftwidth=4
" On pressing tab, insert 4 spaces
set expandtab
```

File: macros/plotSCurveFitResults.py [hunk 4]
```diff
@@ -383,9 +394,19 @@
...
+	    dict_ScurveEffPed[chamberAndScanDatePair][-1].Draw("E1")
         else:
             dict_ScurveEffPed[chamberAndScanDatePair][-1].Draw("sameE1")
+	
+	canvScurveSigma_boxPlot.cd()
+	dict_ScurveSigma_boxPlot[chamberAndScanDatePair].SetFillColorAlpha(getCyclicColor(idx), 0.3)
+	dict_ScurveSigma_boxPlot[chamberAndScanDatePair].SetLineColor(getCyclicColor(idx))
+	if idx == 0:
+	    dict_ScurveSigma_boxPlot[chamberAndScanDatePair].GetXaxis().SetTitle("VFAT position")
+            dict_ScurveSigma_boxPlot[chamberAndScanDatePair].GetYaxis().SetTitle("Noise (fC)")
```
Comment (bdorney) [inline macros/plotSCurveFitResults.py]: `Noise #left(fC#right)`.

**Review by bdorney — CHANGES_REQUESTED** [review]
Slight formatting changes needed.  Otherwise good.

**Comment by bdorney** [comment]
Also it seems your branch `caruta:feature_BoxPlot` is out of date with the central repository.

To resolve this, after having committed all changes to `feature_BoxPlot`, execute:

```bash
git checkout develop
git pull <name of gemdaq repo> develop
git checkout feature_BoxPlot
git rebase develop
```

This will apply your changes, one at a time, onto develop automatically.  If a conflict occurs you'll need to resolve it manually.  You can ask me on Mattermost for how to do this if it occurs.

**Review by bdorney — APPROVED** [review]

**Comment by bdorney** [comment]
Student leaves tomorrow @jsturdy @mexanick @BenjaminRS.  If you could review PR today would be appreciated.  Otherwise I will force merge tomorrow and resolve any discrepancies myself.

**Review by mexanick — APPROVED** [review]

**Pull request merged**